packages/core/core-flows/src/defaults/workflows/create-defaults.ts

```diff
@@ -1,17 +1,23 @@
 import { createWorkflow } from "@medusajs/workflows-sdk"
 import { createDefaultSalesChannelStep } from "../../sales-channel"
+import { createDefaultRegionStep } from "../steps"
 import { createDefaultStoreStep } from "../steps/create-default-store"
 
 export const createDefaultsWorkflowID = "create-defaults"
 export const createDefaultsWorkflow = createWorkflow(
```

Comment:
thought(non-blocking): Might make sense to kill this defaults workflow, except for the store part, in favor of a seed script. There will be many other resources we need to load in by default to prepare a simple store setup, and I thought we might as well make the core less opinionated about which those resources are, if we can.

Reply:
@olivermrbl do you think it might be a good idea for the seed itself to be a workflow? What I am thinking is, rather than having yet another concept for seeding data, you can define your own `create-defaults` workflow. 

Then, we support "hot replacement" of workflows - if you attempt to register a workflow with the same name, the original one gets replaced. I think this replacement mechanism would be useful if eg. you want to use medusa as-is, except `create-products` should do one more step, you can just create a new workflow and that's it, you wouldn't need to touch the HTTP endpoints, for example.

Just thinking out loud, I am also not 100% sure this is a good idea, but it might also be a good intro to workflows for users.

Reply:
> do you think it might be a good idea for the seed itself to be a workflow?

It was definitely the plan to build the default seed script as a workflow. The idea is to call the workflow we have now in a script rather in the core loader.

> rather than having yet another concept for seeding data

I am not sure I understand – as mentioned above this would be a command executing a script like so:
```
// in package.json
"yarn seed": "medusa script ../workflows/defaults.ts"
```

As such, seeding data will not be a separate concept, but something enabled via our CLI + a workflow. If users want to seed their own data, they have the defaults workflow locally in their project and can make the changes directly to it. Thereby, effectively replacing it as you also mention. 

> you want to use medusa as-is, except create-products should do one more step, you can just create a new workflow and that's it, you wouldn't need to touch the HTTP endpoints, for example.

From what I understand, you are talking about extending the flows in `core-flows`, which is definitely on the radar for further improvements to the Workflows SDK. 

Right now, you can, in theory, import the workflow in your own project and append or remove steps. However, we will likely discourage this in favor of dedicated "hooks" in some of the flows we expect to be relevant for extensions, e.g. create product or complete cart. The hooks would be a measure against users messing up the business critical flows.


Nothing is set in stone, but this is my current thinking on seeding defaults data.

Reply:
@olivermrbl yeah I am talking about extending workflows, but also supporting replacement (and not just a purely additive extension). I wasn't aware of the hooks approach we've thought about 👍 

Yeah ok it makes sense for the workflow to be ran outside of the medusa app, I agree. I just think it would be nice if it is a workflow, so you also don't end up with a partial seed data.

Reply:
@olivermrbl I can move this over to the test seeds, but I'm guessing we still keep the changes made in find-one-or-any-region or do we close this PR?

Reply:
> but I'm guessing we still keep the changes made in find-one-or-any-region

Yeah, think we can keep those in 👍